gammapy/catalog/fermi.py

```diff
@@ -660,3 +662,200 @@ def __init__(self, filename=None):
+
+        return ss
+
+    def _get_flux_values(self, prefix='Flux', unit='cm-2 s-1'):
+        if prefix not in ['Flux', 'Unc_Flux']:
+            raise ValueError(
+                "Must be one of the following: 'Flux', 'Unc_Flux'")
+
+        values = [self.data[prefix + _ + 'GeV'] for _ in self._ebounds_suffix]
+        return Quantity(values, unit)
+
+
+    @property
+    def spectral_model(self):
+        """
+        Best fit spectral model `~gammapy.spectrum.models.SpectralModel`.
+        """
+        spec_type = self.data['SpectrumType'].strip()
+        pars = {}
+        pars['amplitude'] = Quantity(
+            self.data['Flux_Density'], 'GeV-1 cm-2 s-1')
+        pars['reference'] = Quantity(self.data['Pivot_Energy'], 'GeV')
+
+        if spec_type == 'PowerLaw':
+            pars['index'] = Quantity(self.data['Spectral_Index'], '')
+            return PowerLaw(**pars)
+
+        elif spec_type == 'LogParabola':
+            pars['alpha'] = Quantity(self.data['Spectral_Index'], '')
+            pars['beta'] = Quantity(self.data['beta'], '')
+            return LogParabola(**pars)
+
+        else:
+            raise ValueError(
+                'Spectral model {} not available'.format(spec_type))
+
+    @property
+    def flux_points(self):
+        """
+        Differential flux points (`~gammapy.spectrum.FluxPoints`).
+        """
+        table = Table()
+        table.meta['SED_TYPE'] = 'flux'
+        e_ref = self._ebounds.log_centers
+        table['e_ref'] = e_ref
+        table['e_min'] = self._ebounds.lower_bounds
+        table['e_max'] = self._ebounds.upper_bounds
+
+        flux = self._get_flux_values()
+        flux_err = self._get_flux_values('Unc_Flux')
+        table['flux'] = flux
+        table['flux_errn'] = np.abs(flux_err[:, 0])
+        table['flux_errp'] = flux_err[:, 1]
+
+        nuFnu = self._get_flux_values('nuFnu', 'erg cm-2 s-1')
+        table['eflux'] = nuFnu
+        table['eflux_errn'] = np.abs(nuFnu * flux_err[:, 0] / flux)
+        table['eflux_errp'] = nuFnu * flux_err[:, 1] / flux
+
+        is_ul = np.isnan(table['flux_errn'])
+        table['is_ul'] = is_ul
+
+        # handle upper limits
+        table['flux_ul'] = np.nan * flux_err.unit
+        table['flux_ul'][is_ul] = table['flux_errp'][is_ul]
+
+        for column in ['flux', 'flux_errp', 'flux_errn']:
+            table[column][is_ul] = np.nan
+
+        # handle upper limits
+        table['eflux_ul'] = np.nan * nuFnu.unit
+        table['eflux_ul'][is_ul] = table['eflux_errp'][is_ul]
+
+        for column in ['eflux', 'eflux_errp', 'eflux_errn']:
+            table[column][is_ul] = np.nan
+
+        # TODO: check if nuFnu is maybe integral flux
+        table['dnde'] = (nuFnu * e_ref ** -2).to('TeV-1 cm-2 s-1')
+        return FluxPoints(table)
+    
+    @property
+    def spectrum(self):
+        """Spectrum model fit result (`~gammapy.spectrum.SpectrumFitResult`)
+
+        TODO: remove!???
```

Comment:
Suggest to remove this TODO comment here and from the other Fermi-LAT catalogs, and instead make a Github issue with Axel, Johannes and me in CC asking the question if the way the catalogs expose the "spectrum fit result" and "energy range" info is good or should be changed / improved.

We're thinking about making the API better, but this TODO comment is not useful for users.

For now you can milestone this one 0.6, and if we don't find something better quickly, we'll move to 0.7.